### PR TITLE
Add database information to getDatabase calls

### PR DESCRIPTION
**Pull request by yuzhu**
Previously, database information are not fetched from the udb and
hence the database information provided to the connnector is missing
some vital information such as owner and location.

### CODE REVIEW

File: table/server/common/src/main/java/alluxio/table/common/udb/UnderDatabase.java [hunk 1]
```diff
@@ -44,4 +46,9 @@
    * @return the {@link UdbContext}
    */
   UdbContext getUdbContext();
+
+  /**
+   * @return get owner name
```
Comment (gpang) [inline table/server/common/src/main/java/alluxio/table/common/udb/UnderDatabase.java]: update comment

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java [hunk 2]
```diff
@@ -118,17 +118,8 @@ public boolean attachDatabase(JournalContext journalContext, String udbType,
               .setDbName(dbName)
               .putAllConfig(map).build()).build());
 
-      try {
-        mDBs.get(dbName).sync(journalContext);
-      } catch (Exception e) {
```
Comment (gpang) [inline table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: Why remove this processing? If we fail to sync the db, we should not have attached the db, right?

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java [hunk 3]
```diff
@@ -435,11 +434,56 @@ public void remove() {
...
+        throw new UnsupportedOperationException(
+            "GetDbInfoIteratorr#Iterator#remove is not supported.");
+      }
+    };
+  }
+
   @Override
   public Iterator<Journal.JournalEntry> getJournalEntryIterator() {
     List<Iterator<Journal.JournalEntry>> componentIters = StreamUtils
         .map(JournalEntryIterable::getJournalEntryIterator, mDBs.values());
```
Comment (gpang) [inline table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: why can't the `mDBs.values()` also add the `DBInfo` entries?

File: table/server/master/src/main/java/alluxio/master/table/Database.java [hunk 4]
```diff
@@ -48,22 +49,21 @@
...
     mContext = context;
     mType = type;
     mName = name;
     mTables = new ConcurrentHashMap<>();
     mUdb = udb;
     mConfig = configMap;
+    if (dbInfo == null) {
+      mDatabaseInfo = new DatabaseInfo();
+    } else {
+      mDatabaseInfo = dbInfo;
```
Comment (gpang) [inline table/server/master/src/main/java/alluxio/master/table/Database.java]: Is there ever a scenario when the `dbInfo` is already populated? Can't we just keep it `null` until it is updated? I think a failed db sync will remove the db. Then, we don't have to pass in `null` as a parameter.

File: table/server/common/src/main/java/alluxio/master/table/DatabaseInfo.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * The database information class.
+ */
+public class DatabaseInfo {
+  private String mLocation;
```
Comment (gpang) [inline table/server/common/src/main/java/alluxio/master/table/DatabaseInfo.java]: We can make all of these `final`?

File: table/server/master/src/test/java/alluxio/master/table/TestDatabase.java [hunk 6]
```diff
@@ -29,6 +32,8 @@
 public class TestDatabase implements UnderDatabase {
   public static final String TEST_UDB_NAME = "test_udb_name";
   public static final String TABLE_NAME_PREFIX = "test_table_name";
+  public static DatabaseInfo sTestDbInfo = new DatabaseInfo("test://test", "TestOwner",
+      PrincipalType.USER, "comment", ImmutableMap.of());
```
Comment (gpang) [inline table/server/master/src/test/java/alluxio/master/table/TestDatabase.java]: can the map contain at least 1 mapping?

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java (hunk 3 above)
Comment (yuzhu) [reply table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: Here we are calling `getJournalEntryIterator` on every db, `mDBs.values` is a list of all DB objects.

File: table/server/master/src/test/java/alluxio/master/table/TestDatabase.java (hunk 6 above)
Comment (yuzhu) [reply table/server/master/src/test/java/alluxio/master/table/TestDatabase.java]: done

File: table/server/master/src/main/java/alluxio/master/table/Database.java (hunk 4 above)
Comment (yuzhu) [reply table/server/master/src/main/java/alluxio/master/table/Database.java]: done

File: table/server/common/src/main/java/alluxio/table/common/udb/UnderDatabase.java (hunk 1 above)
Comment (yuzhu) [reply table/server/common/src/main/java/alluxio/table/common/udb/UnderDatabase.java]: done

File: table/server/common/src/main/java/alluxio/master/table/DatabaseInfo.java (hunk 5 above)
Comment (yuzhu) [reply table/server/common/src/main/java/alluxio/master/table/DatabaseInfo.java]: done

File: table/server/common/src/main/java/alluxio/master/table/DatabaseInfo.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(mLocation, mOwnerName, mOwnerType, mComment, mParameters);
+  }
+
+  @Override
+  public String toString() {
+    return "DatabaseInfo{"
```
Comment (gpang) [inline table/server/common/src/main/java/alluxio/master/table/DatabaseInfo.java]: can we use `MoreObjects.toStringHelper(this)` like we do elsewhere in the codebase?

File: table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java (hunk 3 above)
Comment (gpang) [reply table/server/master/src/main/java/alluxio/master/table/AlluxioCatalog.java]: Yes, so when we are returning journal entries for a database (the table entries), can we also return the dbInfo one as well?

**Comment by yuzhu** [comment]
@gpang addressed comments. PTAL

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by gpang** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**